visualization/app/codeCharta/ui/codeMap/codeMap.arrow.service.ts

```diff
@@ -8,11 +8,11 @@ import { SettingsService } from "../../state/settingsService/settings.service"
 import { ColorConverter } from "../../util/color/colorConverter"
 import { CodeMapBuilding } from "./rendering/codeMapBuilding"
 
-export class CodeMapArrowService implements BuildingHoveredSubscriber, BuildingUnhoveredSubscriber {
+export class CodeMapArrowService
+	implements BuildingSelectedEventSubscriber, BuildingDeselectedEventSubscriber, BuildingHoveredSubscriber, BuildingUnhoveredSubscriber {
 	private VERTICES_PER_LINE = 5
 	private map: Map<String, Node>
```

Comment:
This is not your code, but we should move the map in it's own service/redux. Whenever we build the 3D-map in the preRenderService, we can build this map as well. This map can be quite useful in a lot of other classes as well.